### ISSUE TITLE
make it so you only have to click “enable accessibility” once in a flutter web app

### ISSUE THREAD

**Issue by crentava** [flutter/flutter#77507]
## Use case

I’ve been experimenting with flutter for web for the past couple of days and I’ve noticed that every time a flutter web app is opened and you have a screen reader, you must click the “enable accessibility” button every time. This could quickly become repetitive and may put screen reader users off from using the app
## Proposal

my proposal is to have the user only click “enable accessibility” The first time they visit the page. The app would somehow save that preference in either the browser or somewhere else so that when the same user opens the web app, again they don’t have to keep clicking the option

**Comment by darshankawar** [comment]
May be related https://github.com/flutter/flutter/issues/52104

**Comment by crentava** [comment]
that was the issue that inspired me to make this but it’s not directly related. I learned from that issue that the prompt cannot be disabled because there is no way of detecting screen readers on the web. So I thought why not save the enable accessibility preference so that you don’t have to keep clicking the button

**Comment by yjbanov** [comment]
Accessibility can be enabled programmatically as follows:

```dart
runApp(...); // this should go first to initialize bindings
RendererBinding.instance!.setSemanticsEnabled(true);
```

Saving the accessibility state can be done using `package:shared_preferences` or using your own method (e.g. indexeddb).

I think this could be a great pub package, if anyone is interested in building it.

**Comment by yjbanov** [comment]
Update: unfortunately, the method of enabling accessibility explained in https://github.com/flutter/flutter/issues/77507#issuecomment-794630383 had to be rolled back :(

We will look for ways to fix this.
